src/smart_markers/mod_smart_markers.erl

```diff
@@ -167,11 +211,16 @@ is_valid_host(Acc, From, To) ->
 -spec extract_chat_markers(mongoose_acc:t(), jid:jid(), jid:jid(), exml:element()) ->
 	[chat_marker()].
 extract_chat_markers(Acc, From, To, Packet) ->
-    TS = mongoose_acc:timestamp(Acc),
     case mongoose_chat_markers:list_chat_markers(Packet) of
         [] -> [];
         ChatMarkers ->
-            CM = #{from => From, to => To, thread => get_thread(Packet), timestamp => TS},
+            TS = mongoose_acc:timestamp(Acc),
+            CM = #{from => From,
+                   to => To,
+                   thread => get_thread(Packet),
+                   timestamp => TS,
+                   type => undefined,
+                   id => undefined},
```

Comment:
Why are these set here actually?

Reply:
Performance: https://www.erlang.org/doc/efficiency_guide/maps.html
It's faster to create a map with all the keys it'll need and then modify its values, than to add keys to a map. Small maps are implemented as tuples, and adding keys to the map means having to copy the entire tuple with the keys again 😇

Reply:
So I thought it was about performance 😄
It's a funny case, if the compiler was _really_ clever, it could see what will happen with this map and expand it already...

Reply:
OTP guys are very conservative about the compiler being too clever because it dirties tracing. They're only recently being more aggressive about optimising the jit compiler, more than the traditional interpreter.
Nevertheless I don't know what the compiler would do here, would be interesting to dump the assembly and see, but allow me to do that some other day xD

Reply:
Yeah, sure, just a thought. And I agree, that in some special cases, crazy debugging, it could be a nasty surprise to see some unexpected fields being added by who knows whom. xD